packages/medusa/src/repositories/product.ts

```diff
@@ -298,7 +300,10 @@ export class ProductRepository extends Repository<Product> {
     const entitiesIdsWithRelations = await this.queryProductsWithIds(
       entitiesIds,
       groupedRelations,
-      withDeleted
+      withDeleted,
+      Array.isArray(idsOrOptionsWithoutRelations)
```

Comment:
since you've added another param (order) to queryProductsWithIds, I think you need to add an empty array as selector as that was omitted due to being an optional parameter

Reply:
Nice catch, regarding the free text search I did not touched it. I ll add it as part of that pr

Reply:
I also tried to cleaned up a bit the repo while not make it more complicated :D